### PR TITLE
2019 01 30 work

### PR DESCRIPTION
**Pull request by odscjames**

### CODE REVIEW

**Comment by odscjames** [comment]
Each commit is self contained so check it out one commit at a time! Thank you

**Comment by odscjames** [comment]
/end_collection_store/ new method we need so we can start working on transforms not running until all data is got.

The other 2 new methods record info on how the scrape went, so we can report on API performance for the analysts later.

File: ocdskingfisherprocess/web/app.py [hunk 1]
```diff
@@ -51,11 +51,39 @@ def api_v1():
     return "OCDS Kingfisher APIs V1"
 
 
+@app.route("/api/v1/submit/end_collection_store/", methods=['POST'])
+def api_v1_submit_end_collection_store():
+    api_key = request.headers.get('Authorization', '')[len('ApiKey '):]
+    if not api_key or api_key not in config.web_api_keys:
+        return "ACCESS DENIED", 401
```
Comment (kindly) [inline ocdskingfisherprocess/web/app.py]: Would really like the ApiKey check to be in one place and not repeated in every API fuction. Partly to save repetition but mainly to make sure all the logic is the exactly the same especially as its security related.

File: ocdskingfisherprocess/web/app.py [hunk 2]
```diff
@@ -51,11 +51,39 @@ def api_v1():
...
+
 @app.route("/api/v1/submit/file/", methods=['POST'])
 def api_v1_submit_file():
-    api_key = request.headers.get('Authorization')[len('ApiKey '):]
+    api_key = request.headers.get('Authorization', '')[len('ApiKey '):]
     if not api_key or api_key not in config.web_api_keys:
-        return "ACCESS DENIED"  # TODO proper error
+        return "ACCESS DENIED", 401
 
     # TODO check all required fields are there!
```
Comment (kindly) [inline ocdskingfisherprocess/web/app.py]: If the fields are not there will it mean the database will be in a state it was never intended to be in.

Comment (odscjames) [reply ocdskingfisherprocess/web/app.py]: Yes, that's a general clean up that should be applied to all end points tho. Issued to make sure it's done https://github.com/open-contracting/kingfisher-process/issues/47

File: ocdskingfisherprocess/web/app.py (hunk 1 above)
Comment (odscjames) [reply ocdskingfisherprocess/web/app.py]: Will update now - that's a quick one

**Pull request merged**